src/lib/operators/join_index.cpp

```diff
@@ -37,9 +37,16 @@ bool JoinIndex::supports(const JoinConfiguration config) {
       index_side_table_type = *config.right_table_type;
     }
     if (index_side_table_type == TableType::References && config.join_mode != JoinMode::Inner) {
-      return false;
+      return false;  // non-inner index joins on reference tables are not supported
+    } else if (config.secondary_predicates) {
+      return false;  // multi predicate index joins are not supported
     } else {
-      return !config.secondary_predicates;
+      const auto is_semi_or_anti_join = config.join_mode == JoinMode::Semi ||
+                                        config.join_mode == JoinMode::AntiNullAsFalse ||
+                                        config.join_mode == JoinMode::AntiNullAsTrue;
+      // if a Semi or Anti* join is executed, the left side is outputted by definition. Choosing the index join when
+      // indexes are not available for the right table would not be beneficial.
+      return !(is_semi_or_anti_join && config.index_side == IndexSide::Left);
```

Comment:
@mrks here we do not support the configuration that you mentioned: if we don't have indexes for the right table chunks but a Semi or Anti* join, why should we choose the index join? I would expect the optimizer to choose a different join operator in that case since we cannot benefit from the indexed position lookups.